test/integration/revocation_integration_test.go

```diff
@@ -0,0 +1,308 @@
+		require.Nil(t, crdClient.Create(ctx, secrets[idx]))
+		// Assert that the Kube secret exists + has correct Data.
+		assertSync(t, secrets[idx])
+
+		deleteKV(t, secrets[idx])
+	}
+
+	getTokenData := func(t *testing.T, accessor string) (map[string]interface{}, error) {
+		t.Helper()
+
+		resp, err := vClient.Logical().WriteWithContext(ctx, "auth/token/lookup-accessor", map[string]interface{}{"accessor": accessor})
+		if err != nil || resp == nil {
+			return nil, err
+		}
+		return resp.Data, nil
+	}
+
+	// Get all Vault token accessors that are created from the test by filtering for those that have only the default
+	// and revocationK8sOutputs.PolicyName policies
+	var testTokenAccessors []string
+	resp, err := vClient.Logical().ListWithContext(ctx, "auth/token/accessors")
+	require.NoError(t, err)
+	require.NotNil(t, resp)
+
+	accessors, ok := resp.Data["keys"].([]interface{})
+	require.True(t, ok)
+
+	for _, accessor := range accessors {
+		tokenData, err := getTokenData(t, accessor.(string))
+		if assert.NoError(t, err) {
+			policies := tokenData["policies"].([]interface{})
+			if len(policies) == 2 &&
+				(policies[0] == outputs.PolicyName || policies[1] == outputs.PolicyName) {
+				testTokenAccessors = append(testTokenAccessors, accessor.(string))
+			}
+		}
+	}
+	require.Len(t, testTokenAccessors, len(auths))
+
+	// Get all Vault token secrets of the test auth objects
+	var secretList corev1.SecretList
+	labels := ctrlclient.MatchingLabels{"app.kubernetes.io/component": "client-cache-storage"}
+	opts := []ctrlclient.ListOption{
+		ctrlclient.InNamespace(operatorNS),
+		labels,
+	}
+	crdClient.List(ctx, &secretList, opts...)
+	var testTokenSecrets []corev1.Secret
+	for _, item := range secretList.Items {
+		fmt.Printf("item %s %+v", item.Name, item.Labels)
+		for _, auth := range auths {
+			if val, ok := item.Labels["auth/UID"]; ok && val == string(auth.UID) {
+				testTokenSecrets = append(testTokenSecrets, item)
+				break
+			}
+		}
+	}
+	require.Len(t, testTokenSecrets, len(auths))
+
+	exportKindLogs(t)
+	// Uninstall vso resource
+	terraform.Destroy(t, &terraform.Options{
+		// Set the path to the Terraform code that will be tested.
+		TerraformDir: tfDir,
+		Targets:      []string{"helm_release.vault-secrets-operator"},
+		Vars: map[string]interface{}{
+			"k8s_vault_connection_address": testVaultAddress,
+			"k8s_test_namespace":           testK8sNamespace,
+			"k8s_config_context":           k8sConfigContext,
+			"vault_kvv2_mount_path":        testKvv2MountPath,
+			"operator_helm_chart_path":     chartPath,
+		},
+	})
+
+	// Check if all test tokens were revoked.
+	for _, accessor := range testTokenAccessors {
+		// expect to receive an error that contains "invalid accessor" when looking up the token using its accessor
+		// as an indication that the token was successfully revoked
+		_, err = getTokenData(t, accessor)
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "invalid accessor")
+	}
```

Comment:
I love that we can test all this for real. Nice job!

Reply:
Thank you!